tests/integration/test_tracking.py

```diff
@@ -1116,6 +1116,7 @@ def test_memory_records_are_written(self, tmp_path):
 
         assert memory_records
         assert all(record.rss > 0 for record in memory_records)
+        assert all(record.heap > 0 for record in memory_records)
```

Comment:
Isn't this racy? If a `MemoryRecord` gets written after the `allocator.free()` call and before the `Tracker.__exit__` call, won't it have `record.heap` set to 0? Likewise for one written after `Tracker.__enter__` and before `allocator.valloc()`

Reply:
I changed this to check if a heap field equals the heap size at some point, which is less racy and probably more aligned to what we want to check here.